lib/puppet/provider/package/pkgng.rb

```diff
@@ -0,0 +1,145 @@
+require 'puppet/provider/package'
+
+Puppet::Type.type(:package).provide :pkgng, :parent => Puppet::Provider::Package do
+  desc "A PkgNG provider for FreeBSD and DragonFly."
+
+  commands :pkg => "/usr/local/sbin/pkg"
+
+  confine :operatingsystem => [:freebsd, :dragonfly]
+  confine :pkgng_enabled => :true
+
+  defaultfor :operatingsystem => :freebsd
+  defaultfor :pkgng_enabled => :true
```

Comment:
I'd rather not have this exist as a core fact because it's basically provider logic.  If we can guarantee that the provider defaulting mechanism is client-side (because this logic executes child processes), I see no reason why we can't extend `defaultfor` to support dynamic checks via `Proc`s. That way the provider can keep this "should I be the default provider?" in-house.